### PR TITLE
LocktimeSoftware.NetLimiter4 version 4.1.10.0

### PR DESCRIPTION
**Pull request by krystofkrticka**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/16625)

### CODE REVIEW

**Comment by krystofkrticka** [comment]
Weird it installs fine with switches I added. Now installer only opens CMD window to install to Windows service.

**Comment by krystofkrticka** [comment]
> Weird it installs fine with switches I added. Now installer only opens CMD window to install Windows service.

Right now I checked one again if this package install correctly and it install just fine so that's why I don't get why installation failed

**Comment by denelon** [comment]
@krystofkrticka we are getting a 1603 error in our logs.

**Comment by denelon** [comment]
Did you test this under user or administrator context? Did you try with PowerShell or CMD? Can you share the output of `winget --info`?

**Comment by krystofkrticka** [comment]
> Did you test this under user or administrator context? Did you try with PowerShell or CMD? Can you share the output of `winget --info`?

Sorry I was unable to answer earlier but this message came to my Inbox at 1:20 am CEST.

So I tried to install package using elevated powershell and even cmd and in both cases it installed just fine.
When installing from non elevated powershell or cmd I just get UAC prompt to elevate installer.

Here is output of `winget --info`

```
Windows Package Manager v1.0.11451
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19043.1023
Package: Microsoft.DesktopAppInstaller v1.11.11451.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links               
--------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
```

**Comment by krystofkrticka** [comment]
> > Did you test this under user or administrator context? Did you try with PowerShell or CMD? Can you share the output of `winget --info`?
> 
> Sorry I was unable to answer earlier but this message came to my Inbox at 1:20 am CEST.
> 
> So I tried to install package using elevated powershell and even cmd and in both cases it installed just fine.
> When installing from non elevated powershell or cmd I just get UAC prompt to elevate installer.
> 
> Here is output of `winget --info`
> 
> ```
> Windows Package Manager v1.0.11451
> Copyright (c) Microsoft Corporation. All rights reserved.
> 
> Windows: Windows.Desktop v10.0.19043.1023
> Package: Microsoft.DesktopAppInstaller v1.11.11451.0
> 
> Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
> 
> Links               
> --------------------------------------------------------
> Privacy Statement   https://aka.ms/winget-privacy
> License Agreement   https://aka.ms/winget-license
> Third Party Notices https://aka.ms/winget-3rdPartyNotice
> Homepage            https://aka.ms/winget
> ```

In the end every time I try to install this package I end up with successful installation as you can see in this screenshot
image

**Comment by denelon** [comment]
@krystofkrticka no worries on the delay. I've asked the engineering team to take a look.

**Comment by denelon** [comment]
@krystofkrticka are there any prerequisites/dependencies for this package? Did you test in the Sandbox?

**Comment by krystofkrticka** [comment]
> @krystofkrticka are there any prerequisites/dependencies for this package? Did you test in the Sandbox?

Yes, I was testing this in VM

**Comment by ChuckFerring** [comment]
I ran the installer with logging using the command line: .\netlimiter-4.1.10.0.exe /exenoui /qn /l*v "nl-inst-log.txt.
It looks like the service fails to start during install and the installation gets rolled back. Here's a snippet from the log:

MSI (s) (94:CC) [11:42:26:283]: Executing op: CustomActionSchedule(Action=CA_SVC_INSTALL_64,ActionType=1042,Source=C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe,Target=/afterinstall,)
MSI (s) (94:CC) [11:42:27:267]: Note: 1: 1722 2: CA_SVC_INSTALL_64 3: C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe 4: /afterinstall 
CustomAction CA_SVC_INSTALL_64 returned actual error code -532462766 (note this may not be 100% accurate if translation happened inside sandbox)
MSI (s) (94:CC) [11:42:27:267]: Product: NetLimiter 4 -- Error 1722. There is a problem with this Windows Installer package. A program run as part of the setup did not finish as expected. Contact your support personnel or package vendor.  Action CA_SVC_INSTALL_64, location: C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe, command: /afterinstall 

When running with the UI everything installs as expected and the service starts. It is not clear to me why the installer doesn't work without the UI.

**Comment by krystofkrticka** [comment]
> I ran the installer with logging using the command line: .\netlimiter-4.1.10.0.exe /exenoui /qn /l*v "nl-inst-log.txt.
> It looks like the service fails to start during install and the installation gets rolled back. Here's a snippet from the log:
> 
> MSI (s) (94:CC) [11:42:26:283]: Executing op: CustomActionSchedule(Action=CA_SVC_INSTALL_64,ActionType=1042,Source=C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe,Target=/afterinstall,)
> MSI (s) (94:CC) [11:42:27:267]: Note: 1: 1722 2: CA_SVC_INSTALL_64 3: C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe 4: /afterinstall
> CustomAction CA_SVC_INSTALL_64 returned actual error code -532462766 (note this may not be 100% accurate if translation happened inside sandbox)
> MSI (s) (94:CC) [11:42:27:267]: Product: NetLimiter 4 -- Error 1722. There is a problem with this Windows Installer package. A program run as part of the setup did not finish as expected. Contact your support personnel or package vendor. Action CA_SVC_INSTALL_64, location: C:\Program Files\Locktime Software\NetLimiter 4\NLSvc.exe, command: /afterinstall
> 
> When running with the UI everything installs as expected and the service starts. It is not clear to me why the installer doesn't work without the UI.
 
@ChuckFerring 
Well that's weird because when I run installer using command line using switches /exenoui /qn service starts just fine.

**Comment by krystofkrticka** [comment]
So I run it too with logging and here is my log
[nl-inst-log.txt](https://github.com/microsoft/winget-pkgs/files/6640780/nl-inst-log.txt)

**Comment by krystofkrticka** [comment]
> Hello @krystofkrticka,
> The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

Well this didn't help

**Comment by krystofkrticka** [comment]
@ChuckFerring @denelon This is really strange behavior because I tried to install this application using manifest on multiple windows 10 instances both stable and insider in dev ring and always installation was successful and service started.

**Comment by krystofkrticka** [comment]
@ItzLevvie2 Why did you closed this? I was waiting if someone responds to my last comment because problem showing in automated testing was not showing in manual testing.

**Comment by ItzLevvie** [comment]
Closing PRs that haven't had any author response for x amount of days, seems like I'm unable to install the application too (fails on both admin and non-admin) - feel free to reopen otherwise:
image

**Comment by krystofkrticka** [comment]
This is really weird. On my VMs It install every time when I use manifests from my manifest folder created by wingetcreate command.

**Comment by krystofkrticka** [comment]
> This is really weird. On my VMs It installs every time when I use manifests from my manifest folder created by wingetcreate command.

I even tried fresh installed VM.

**Comment by krystofkrticka** [comment]
> Closing PRs that haven't had any author response for x amount of days.
> 
> It seems like I'm unable to install the application too; it fails on both with administrator privileges and without any administrator privileges on a clean, fresh install virtual machine with everything uninstalled - feel free to reopen the PR if this was an error on my end:
> image
> 
> You'll have to bring this issue up with the application's developers.

https://user-images.githubusercontent.com/37749100/124654164-81dee680-de9e-11eb-83dc-25d40bd0205d.mp4

Here is screen recording of successful installation using manifest which I pulled directly from my forked repo. I made this VM specifically to show that it installs fine to me. Tested on freshly installed VM which was just updated to have latest cumulative update.

**Comment by krystofkrticka** [comment]
BTW I can't reopen this PR it just says delete branch. I would have to open new one.

**Comment by ItzLevvie** [comment]
Did another try today, and it works fine now; though I believe this application can't be added until all of the automated tests have successfully passed.

File: manifests/l/LocktimeSoftware/NetLimiter4/4.1.10.0/LocktimeSoftware.NetLimiter4.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Created using wingetcreate
+
+PackageIdentifier: LocktimeSoftware.NetLimiter4
+PackageVersion: 4.1.10.0
+PackageLocale: en-US
+Publisher: Locktime Software
+PackageName: NetLimiter 4
```
Comment (ItzLevvie) [inline manifests/l/LocktimeSoftware/NetLimiter4/4.1.10.0/LocktimeSoftware.NetLimiter4.locale.en-US.yaml:7]:
```suggestion
PackageName: NetLimiter
```

**Comment by krystofkrticka** [comment]
@ItzLevvie I can't commit this suggestion because PR is closed

**Comment by krystofkrticka** [comment]
I found out that too that MSI packages sometimes don't want to install ending with 1603 error, for me this behavior is usually fixed by rebooting Windows.

**Comment by ItzLevvie** [comment]
> @ItzLevvie I can't commit this suggestion because PR is closed. Should I make new PR?

Sure; it's worth a try. If it fails validation, there's nothing I can do and cannot approve the PR either because Community Moderators can only merge if there's an `Azure-Pipeline-Passed` label on the PR.

You'll either have to talk to the Windows Package Manager team it or just close the PR and come back to it at a later time in the future.

**Comment by krystofkrticka** [comment]
@ItzLevvie OK, I'll manually edit it and submit new PR to see if it passes this time.

**Comment by krystofkrticka** [comment]
OK tried to create new PR with newer version which came out few days ago. #20129

**Comment by krystofkrticka** [comment]
@denelon I'm sorry for saying that there wasn't any dependency problem was that there is VC++ dependency but I didn't caught it because aditional tools from my virtualization SW installed them.

**Comment by krystofkrticka** [comment]
> there aren't any dependencies

I take this back. As i wrote in my previous [comment](https://github.com/microsoft/winget-pkgs/pull/16625#issuecomment-876609034)

**Comment by denelon** [comment]
@krystofkrticka no worries. Dependencies has been a very complex area to navigate. We even ran into some of the same kinds of issues when we were setting our testing infrastructure up. Some of the testing tools added dependencies that wouldn't have existed in a true "clean" install, so we had some challenges making sure the environment wasn't contaminated with things that exist outside of a system in a default configuration.

**Pull request closed without merging**